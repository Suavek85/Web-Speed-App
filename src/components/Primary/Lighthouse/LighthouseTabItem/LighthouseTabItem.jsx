import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames/bind'

import { tabsHeader } from '../../../../constants/content'
import styles from './LighthouseTabItem.scss'

export default function LighthouseTabItem(props) {
  const { activeTab } = props

  //STYLE
  const cx = classNames.bind(styles)
  
  const geTabItemClass = (tabIndex, activeTabIndex) => cx({
    block: true,
    blockActive: activeTabIndex == tabIndex,
  })

  return (
    tabsHeader.map((el,i) => ( <div 
      id={i} 
      key={i} 
      className={ geTabItemClass(i, activeTab) }>
      {el.content}
    </div>) )
  )
}

LighthouseTabItem.propTypes = {
  activeTab: PropTypes.string,
}