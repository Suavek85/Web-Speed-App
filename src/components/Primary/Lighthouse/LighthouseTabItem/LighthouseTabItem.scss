@import '../../../../scss/styles.scss';

.lighthouse__tabsitem {
    display: flex;
    justify-content: center;
    align-items: center;
    height: inherit;
    background-color: $color-beige;
    width: 100%;
    text-align: center;
    cursor: pointer;
    padding: 0px 5px;
    font-size: 12px;
    &:hover {
      background-color: $color-grey;
      opacity: .8;
      color: $color-white;
    }
    &--active {
      background-color: $color-grey;
      color: $color-white;
    }
  
  }

  :export {
    blockTabsItem: lighthouse__tabsitem;
    blockTabsItemActive: lighthouse__tabsitem--active;
}