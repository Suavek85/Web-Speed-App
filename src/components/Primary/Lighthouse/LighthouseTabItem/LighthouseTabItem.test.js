import React from 'react'
import { shallow } from 'enzyme'
import LighthouseTabItem from './LighthouseTabItem'

describe('LighthouseTabItem snapshot', () => {

  it('should match the snapshot', () => {
    const wrapper = shallow(<LighthouseTabItem />)
    expect(wrapper).toMatchSnapshot()
  })

})