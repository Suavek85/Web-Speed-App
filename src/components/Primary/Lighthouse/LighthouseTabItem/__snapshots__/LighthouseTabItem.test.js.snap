// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LighthouseTabItem snapshot should match the snapshot 1`] = `
Array [
  <div
    className="block"
    id={0}
    key="0"
  >
    First Paint
  </div>,
  <div
    className="block"
    id={1}
    key="1"
  >
    Largest Paint
  </div>,
  <div
    className="block"
    id={2}
    key="2"
  >
    Speed Index
  </div>,
  <div
    className="block"
    id={3}
    key="3"
  >
    Interactive
  </div>,
  <div
    className="block"
    id={4}
    key="4"
  >
    Blocking Time
  </div>,
  <div
    className="block"
    id={5}
    key="5"
  >
    Layout Shift
  </div>,
]
`;
